CHANGELOG.md

```diff
@@ -0,0 +1,140 @@
+- New editor theme, with customizable base color, highlight color and contrast.
+  - A light editor theme option is now available, with icons suited to light backgrounds.
+  - Alternative dark gray and Arc colors are available out of the box.
+- New adaptive text editor theme, adjusting automatically based on the editor colors.
+- Support for macOS trackpad gestures in the editor.
+- Exporting to macOS now creates a `.dmg` disk image if exporting from an editor running on macOS.
+  - Signing the macOS export now is possible if running macOS (requires a valid code signing certificate).
+- Exporting to Windows now changes the exported project's icon using `rcedit` (requires WINE if exporting from Linux or macOS).
+- Improved build system.
+  - Support for compiling using Visual Studio 2017.
+  - [SCons](http://scons.org/) 3.0 and Python 3 are now supported (SCons 2.5 and Python 2.7 still work).
+  - Link-time optimization can now be enabled by passing `use_lto=yes` to the SCons command line.
+    - Produces faster and sometimes smaller binaries.
+    - Currently only supported with GCC and MSVC.
+  - Added a progress percentage when compiling Godot.
+  - `.zip` archives are automatically created when compiling HTML5 export templates.
+- Easier and more powerful way to create editor plugins with EditorPlugin and related APIs.
+
+### Changed
+
+- Increased the default low-processor-usage mode FPS limit (60 → 125).
+  - This makes the editor smoother and more responsive.
+- Increased the default 3D editor camera's field of view (55 → 70).
+- Increased the default 3D Camera node's field of view (65 → 70).
+- Changed the default editor font (Droid Sans → [Noto Sans](https://www.google.com/get/noto/)).
+- Changed the default script editor font (Source Code Pro → [Hack](http://sourcefoundry.org/hack/))
+- Renamed `engine.cfg` to `project.godot`.
+  - This allows users to open a project by double-clicking the file if Godot is associated to `.godot` files.
+- Some methods from the `OS` singleton were moved to the new `Engine` singleton.
+- Switched from [GLEW](http://glew.sourceforge.net/) to [GLAD](http://glad.dav1d.de/) for OpenGL wrapping.
+- Changed the SCons build flag for simple logs (`colored=yes` → `verbose=no`).
+- The HTML5 platform now uses WebGL 2.0 (instead of 1.0).
+- Redesigned the Godot logo to be more legible at small sizes.
+
+### Deprecated
+
+- `opacity` and `self_opacity` are replaced by `modulate` and `self_modulate` in all 2D nodes, allowing for full color changes in addition to opacity changes.
+
+### Removed
+
+- Skybox support.
+  - Replaced with panorama skies, which are easier to import.
+- Opus audio codec support.
+  - This is due to the way the new audio engine is designed.
+- HTML5 export using asm.js.
+  - Only WebAssembly is supported now, since all browsers supporting WebGL 2.0 also support WebAssembly.
+
+[3.0]: https://github.com/godotengine/godot/compare/2.1.4-stable...3.0-stable
```

Comment:
I don't think this will provide a meaningful comparison, as 2.1.4-stable and 3.0-stable come from different branches. `2.1-stable...3.0-stable` would be better (also different branches strictly speaking as 2.1-stable was tagged after branching `2.1`, but it should be easier for git to compare them).